### PR TITLE
Develop/fixes/dlsv2 656 text area height no js 2

### PR DESCRIPTION
**Pull request by martinfoster-hee**
### JIRA link
https://hee-dls.atlassian.net/jira/software/projects/DLSV2/boards/1?selectedIssue=DLSV2-656

### Description
Swapped the standard textarea controls to use textarea view component. This now correctly renders a 5 row textarea when javascript is disabled in the browser (with js enabled the control still renders as a textarea with the html-editor functionality).

The new control appears on the following four pages:
AssessmentQuestionsOptions.cshtml
Description.cshtml
Competency.cshtml
CompetencyGroup.cshtml

### Screenshots
See attached.

-----
### Developer checks
I have:
- [x] Run the formatter and made sure there are no IDE errors (see [info on Text Editor settings](https://hee-tis.atlassian.net/wiki/spaces/TP/pages/3546185813/DLS+Dev+Process) to avoid whitespace changes)
- [x] Written tests for the changes (accessibility tests, unit tests for controller, data services, services, view models, etc)
- [x] Manually tested my work with and without JavaScript
- [x] Tested any Views or partials created or changed with [Wave Chrome plugin](https://chrome.google.com/webstore/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh/related) and addressed any valid accessibility issues
- [ ] Updated/added documentation in [Confluence](https://hee-tis.atlassian.net/wiki/spaces/TP/pages/3546939432/DLS+Code) and/or [GitHub Readme](https://github.com/TechnologyEnhancedLearning/DLSV2/blob/master/README.md). List of documentation links added/changed:
  - [doc_1_here](link_1_here)
- [x] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken
- [x] Scanned over my own MR to ensure everything is as expected and it looks right in the browser

final_assessement_outcome_js_desktop
final_assessement_outcome_nojs_desktop
edit_competency_group_js_desktop
edit_competency_group_nojs_desktop
edit_competency_js_desktop
edit_competency_nojs_desktop
edit_framework_description_js_desktop
edit_framework_description_nojs_desktop

### CODE REVIEW

**Review by kevwhitt-hee — CHANGES_REQUESTED** [review]
This is working well when I run it in dev. See comment about moving value to set functions out to a helper class. This would allow us to call them from other view components if we face the same problem.

File: DigitalLearningSolutions.Web/ViewComponents/TextAreaViewComponent.cs [hunk 1]
```diff
@@ -49,5 +58,33 @@ public class TextAreaViewComponent : ViewComponent
                 characterCount);
             return View(textBoxViewModel);
         }
+
+        private string ValueToSetForSimpleType(object model, string aspFor, bool populateWithCurrentValue, out IEnumerable<string> errorMessages)
```
[the remark this answers is not in the document]
Comment (martinfoster-hee) [reply DigitalLearningSolutions.Web/ViewComponents/TextAreaViewComponent.cs:62]: Yes good idea - we are likely to need this for refactoring other types of view components where we have complex view models.

[the remark this answers is not in the document]
Comment (martinfoster-hee) [reply DigitalLearningSolutions.Web/ViewComponents/TextAreaViewComponent.cs]: @kevwhitt-hee I've refactored the two model binding functions to a helper class as requested. 

I've also refactored the main calling function which decides which of the two methods to call as that will reduce code simplify the calls in the controllers (although the controllers can still call the other two functions directly if needed).

**Review by kevwhitt-hee — APPROVED** [review]
Looks good. Tests for the helper are a good addition too.

**Pull request merged**